### PR TITLE
Update GlobalStrain documentation

### PR DESCRIPTION
**Pull request by SudiptaBiswas**
- Adds requirements to the tests
- Updates the global strain documentation 

Refs #11314 and #7387

### CODE REVIEW

**Comment by SudiptaBiswas** [comment]
Ping @bwspenc

**Comment by brianmoose** [comment]
#12212 merged in some symbolic links that seem to break the clang format check.
You will probably need to rebase on the devel branch and repush

File: modules/tensor_mechanics/doc/content/syntax/Modules/TensorMechanics/GlobalStrain/index.md [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
+
+Here, indices $i$ and $j$ corresponds to the periodic directions and associated periodic displacement components, respectively. $\sigma^0_{ij}$ denotes the components of the stress tensor representing the applied loads in the periodic directions. The components of the global strain are considered constant over the whole domain.
+
+The total strain $\epsilon_{ij}^{total}$ at any point is calculated considering the contribution of the global strain tensor as,
+
+\begin{equation}
+\epsilon_{ij}^{total} =  \epsilon_{ij} + \epsilon_{ij}^{g} - \epsilon_{ij}^{0}\, \label{strain_eq}
+\end{equation}
+
+where $\epsilon_{ij}$ are the [strain components](/ComputeSmallStrain.md) calculated from the displacement gradients; $\epsilon_{ij}^g$ are the global strain components; and $\epsilon_{ij}^0$ are the eigen strain components. Additional [displacement field](/GlobalDisplacementAux.md) $\boldsymbol u^g$ is generated to visualize the effect of the global strain as
```
Comment (dschwen) [inline modules/tensor_mechanics/doc/content/syntax/Modules/TensorMechanics/GlobalStrain/index.md]: _An_ additional displacement field.

Comment (dschwen) [reply modules/tensor_mechanics/doc/content/syntax/Modules/TensorMechanics/GlobalStrain/index.md]: and use , rather than ; here

**Review by dschwen — DISMISSED** [review]
Thanks for adding verbose docs! The Pronghorn failure is unrelated (due to #12211)

File: modules/tensor_mechanics/test/tests/global_strain/tests [hunk 2]
```diff
@@ -1,44 +1,70 @@
 [Tests]
+  design = 'GlobalStrain/index.md'
+  issues = '#11314'
   [./test]
     type = 'Exodiff'
     input = 'global_strain.i'
     exodiff = 'global_strain_out.e'
+    requirement = "The globalstrain system shall check the volume change due to
+                   applied stress while still maintaining periodicity in 2D."
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/global_strain/tests]: Thanks for adding this stuff!  Just a minor thing:  The requirements should refer to what the code should be able to do and how it should behave, and the test just confirms that it behaves that way.

I would reword this as something like:

The globalstrain system shall correctly compute the volume change due to applied stress while still maintaining periodicity in 2D.

The same principle applies to the other tests that you've written requirements for.

**Review by permcody — APPROVED** [review]
